### PR TITLE
feat(core): append page and page_size to the query parameters in swagger.json

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Append the `page` and `page_size` to the query parameters in the `swagger.json` when the routes use `pagination`

---

Check the pagination middleware like the guard middleware:

- `isPaginationMiddleware`:

    image

- `isGuardMiddleware`:

    image

## Linear Issue Reference

LOG-2856

## Testing

- UTs

    image

- GET /api/swagger.json

    image

- API Reference page

    image

### CODE REVIEW

File: packages/core/src/middleware/koa-pagination.ts [hunk 1]
```diff
@@ -67,4 +78,6 @@ export default function koaPagination<StateT, ContextT, ResponseBodyT>({
       ctx.append('Link', buildLink(ctx.request, page + 1, 'next'));
     }
   };
+
+  return paginationMiddleware;
```
Comment (wangsijie) [inline packages/core/src/middleware/koa-pagination.ts:82]: What is this variable use for?

Comment (gao-sun) [reply packages/core/src/middleware/koa-pagination.ts:82]: i think it's for function name

Comment (gao-sun) [reply packages/core/src/middleware/koa-pagination.ts:82]: i did not expect this (i was thinking arrow function has no "_name"). maybe we can use `instanceof`?

Comment (IceHe) [reply packages/core/src/middleware/koa-pagination.ts:82]: It's used to identify the pagination middleware.

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/middleware/koa-pagination.ts (hunk 1 above)
Comment (IceHe) [reply packages/core/src/middleware/koa-pagination.ts:82]: Added a comment to note it

```ts
  // Name this anonymous function for the utility function `isPaginationMiddleware` to identify it
  const paginationMiddleware: MiddlewareType<
    StateT,
    WithPaginationContext<ContextT>,
    ResponseBodyT
  > = async (ctx, next) => {

```

**Comment by IceHe** [comment]
Rebased master and ddded a comment to note:

```ts
  // Name this anonymous function for the utility function `isPaginationMiddleware` to identify it
  const paginationMiddleware: MiddlewareType<
    StateT,
    WithPaginationContext<ContextT>,
    ResponseBodyT
  > = async (ctx, next) => {
```

**Pull request merged**